x-pack/plugins/cloud/public/types.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+
+export interface CloudStart {
+  /**
+   * A React component that provides a pre-wired `React.Context` which connects components to Cloud services.
+   */
+  CloudContextProvider: FC<{}>;
+  /**
+   * `true` when Kibana is running on Elastic Cloud.
+   */
+  isCloudEnabled: boolean;
+  /**
+   * Cloud ID. Undefined if not running on Cloud.
+   */
+  cloudId?: string;
+  /**
+   * The full URL to the deployment management page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  deploymentUrl?: string;
+  /**
+   * The full URL to the user profile page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  profileUrl?: string;
+  /**
+   * The full URL to the organization management page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  organizationUrl?: string;
+  /**
+   * The full URL to the elasticsearch cluster.
+   */
+  elasticsearchUrl?: string;
+  /**
+   * The full URL to the Kibana deployment.
+   */
+  kibanaUrl?: string;
+}
+
+export interface CloudSetup {
+  /**
+   * Cloud ID. Undefined if not running on Cloud.
+   */
+  cloudId?: string;
+  /**
+   * The deployment's ID. Only available when running on Elastic Cloud.
+   */
+  deploymentId?: string;
+  /**
+   * This value is the same as `baseUrl` on ESS but can be customized on ECE.
+   */
+  cname?: string;
+  /**
+   * This is the URL of the Cloud interface.
+   */
+  baseUrl?: string;
+  /**
+   * The full URL to the deployment management page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  deploymentUrl?: string;
+  /**
+   * The full URL to the user profile page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  profileUrl?: string;
+  /**
+   * The full URL to the organization management page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  organizationUrl?: string;
+  /**
+   * This is the path to the Snapshots page for the deployment to which the Kibana instance belongs. The value is already prepended with `deploymentUrl`.
+   */
+  snapshotsUrl?: string;
+  /**
+   * The full URL to the elasticsearch cluster.
+   */
+  elasticsearchUrl?: string;
+  /**
+   * The full URL to the Kibana deployment.
+   */
+  kibanaUrl?: string;
+  /**
+   * {host} from the deployment url https://<deploymentId>.<application>.<host><?:port>
+   */
+  cloudHost?: string;
+  /**
+   * {port} from the deployment url https://<deploymentId>.<application>.<host><?:port>
+   */
+  cloudDefaultPort?: string;
```

Comment:
Naming is hard. I went with those, but I'm open to better suggestions (or better TSDoc)